staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go

```diff
@@ -105,7 +113,17 @@ func (r *proxyHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	// write a new location based on the existing request pointed at the target service
 	location := &url.URL{}
 	location.Scheme = "https"
-	location.Host = handlingInfo.destinationHost
+	if r.router != nil {
+		ctx = genericapirequest.WithNamespace(ctx, handlingInfo.namespace)
+		rloc, _, err := r.router.ResourceLocation(ctx, handlingInfo.serviceName)
```

Comment:
So it should now be based on listers, which I believe should not be as expensive. I believe caching here is a bad idea. This is seems more critical than services/proxy which always uses an up to date list, I assume because you do not want to risk sending to a stale location. 

I would not term this as sometimes ignoring the destinationHost parameter. That parameter is the cluster IP, if you set the Aggregator to route to the endpoint IP it should always ignore that parameter. If you do not configure it to route, then it should not ignore the parameter. It is behaving the way you configured it.